portal-bridge/src/census/network.rs

```diff
@@ -438,17 +430,15 @@ impl NetworkManager {
             }
         };
 
-        let mut new_peers = 0;
+        let starting_peers = self.network.peers.len();
         for enr in enrs {
-            if self.network.liveness_check(enr).await == LivenessResult::Pass {
-                new_peers += 1;
-            }
+            self.network.liveness_check(enr).await;
         }
-
-        let total_peers = self.network.peers.len();
+        let ending_peers = self.network.peers.len();
+        let new_peers = ending_peers - starting_peers;
```

Comment:
I am assuming the possibility that `new_peers` can be negative as a feature to display we lost peers? Because isn't there a case where `ending_peers` can be less then `starting_peers` i.e. if more peers were removed then added?

This doesn't really look like an issue, just wanted to point it out.

Reply:
I was thinking about it and concluded that it shouldn't happen. Unless there is some weird race-condition and/or Census is performing very badly.

But, you are right, it's technically possible. It's also possible that we change logic in the future and forget to update and this would panic (or difference no longer means that these are added).

So I updated the code to not panic and changed the message.